### DISCUSSION TITLE
问一个uniapp + vue3 + ts在vscode上开发体验的问题

### DISCUSSION THREAD

**Discussion by loclink** [nutui-uniapp/nutui-uniapp#61] [General]
之前一直用的Taro+React+ts开发微信小程序，无论是类型推导或者代码补全都非常舒服，开发过程也非常丝滑。
最近公司有uniapp项目的需求，换过来以后也是在vscode上为了开发体验折腾了很久，目前为止我自己搭的这个环境该有的提示都有了，但还有一个未解决的问题，不太清楚其他同样也在使用vscode开发uniapp + vue3 + ts项目的朋友有没有遇到这个问题：

> 问题描述：项目中我新建了一个`.vue`文件，这个文件就失去了全局组件以及全局对象相关的类型提示，例如`nutui`相关的所有组件以及全局对象`uni`。除非重启vscode，否则这个文件就一直不会有提示。vscode插件使用的是`Volar`和`Typescript Vue Plugin`，同时也禁用了这个工作区的内置Typescript服务
<img width="317" alt="image" src="https://github.com/nutui-uniapp/nutui-uniapp/assets/106688231/095a6e7b-4a7b-4fa7-b40b-59ff9fbeb497">

我的tsconfig.json:

```json
{
  "compilerOptions": {
    "target": "ES2020",
    "module": "ESNext",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "useDefineForClassFields": true,
    "skipLibCheck": true,
    "strict": true,
    /* Node mode */
    "moduleResolution": "Node",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "preserve",

    /* Custom */
    "baseUrl": ".",
    "paths": {
      "@/*": ["src/*"]
    },
    "types": [
      "node",
      "@uni-helper/uni-app-types",
      "@dcloudio/types",
      "./components.d.ts",
      "uniapp-nutui/global.d.ts",
      "wm-uni-design/components.d.ts",
      "vite/client"
    ]
  },
  "include": ["src/**/*", "types"],
  "exclude": ["node_modules"],
  "vueCompilerOptions": {
    "target": 3,
    "nativeTags": ["block", "component", "template", "slot"]
  },
  "references": [{ "path": "./tsconfig.node.json" }]
}
```
这对我来说算是一个比较影响开发体验的问题，希望能得到大佬的解答，万分感谢🙏

**Comment by yang1206** [comment]
这个问题我也遇到了，应该是某个版本的 volar 更新后才有的

**Reply by loclink** [reply]
感谢大佬的回复，因为前段时间都在写react，不太清楚vue现在是个什么情况🙏
